dig_test.go

```diff
@@ -919,6 +919,48 @@ func TestGroups(t *testing.T) {
 		}), "invoke failed")
 	})
 
+	t.Run("groups are provided via option", func(t *testing.T) {
+		c := New(setRand(rand.New(rand.NewSource(0))))
+
+		provide := func(i int) {
+			require.NoError(t, c.Provide(func() int {
+				return i
+			}, Group("val")), "failed to provide ")
+		}
+
+		provide(1)
+		provide(2)
+		provide(3)
+
+		type in struct {
+			In
+
+			Values []int `group:"val"`
+		}
+
+		require.NoError(t, c.Invoke(func(i in) {
+			assert.Equal(t, []int{2, 3, 1}, i.Values)
+		}), "invoke failed")
+	})
+
+	t.Run("group options may not be provided for result structs", func(t *testing.T) {
+		c := New(setRand(rand.New(rand.NewSource(0))))
+
+		type out struct {
+			Out
+
+			Value int `group:"val"`
+		}
+
+		provide := func(i int) {
+			require.Error(t, c.Provide(func() out {
+				return out{Value: i}
+			}, Group("val")), "This Provide should fail")
+		}
+
```

Comment:
function unnecessary here since this is called exactly once. consider inlining it.

```
require.Error(t, c.Provide(...))
```

The contents of the function can also mark the test as failed to guarantee that it's never called.

```
func() out {
  t.Fatal("this function should never be called")
  return out{}
}
```